docs/migration-v2.md

```diff
@@ -0,0 +1,71 @@
+---
+hide_title: true
+title: Migration to v2
+---
+
+# Migration to Remirror v2
+
+This guide lays out the required steps to migrate from Remirror v1 to Remirror v2.
+
+Consult the [announcement post](/blog/announcement-v2) to learn more about the v2 release.
+
+## Installing
+
+If you are using TypeScript in your project, please remove all `@types/prosemirror-*` dependencies from your `package.json` file, for example, `@types/prosemirror-model` or `@types/prosemirror-view`.
+
+After that, you can update all Remirror packages to the latest version. We recommend using the [`npm-check-updates`](https://www.npmjs.com/package/npm-check-updates) package to do this.
+
+```bash
+$ npx npm-check-updates --filter '/remirror|prosemirror/' --deep --upgrade --target greatest
+```
+
+The command above will update all Remirror packages to the latest version in your `package.json` file. Then you can install them with `npm install`, `yarn install`, or `pnpm install`.
+
+## Deduplicate dependencies
+
+Remirror is a monorepo with multiple packages, with which some package managers struggle nowadays. You could therefore end up with duplicate versions of the same package in your project. We recommend you to check your lockfile (usually `package-lock.json`, `yarn.lock` or `pnpm-lock.yaml`) if you have installed multiple versions of the same package. Look out especially for the following packages:
+
+- `prosemirror-state`
+- `prosemirror-model`
+- `@remirror/core`
+- `@remirror/pm`
+
+If you find any duplicate versions, the easiest way is to remove the lockfile and the `node_modules` directory, then install your dependencies again. However, this will update **all** packages in your project, and you might not want to do that.
```

Comment:
We should make the warning much more prominent here. Make this a "if all else fails" approach as this will likely have unintended consequences to other parts of your code base.

The approach above is much less destructive to the lockfile, I think we should be more explicit. 

> Look in your `package.json` for the following. Make sure to note them down, along with whether they are dependencies or _**dev**_ dependencies
>
> * `remirror`
> * `@remirror` scoped packages i.e. `@remirror/pm`
> * `jest-remirror` or `jest-prosemirror`
> * `prosemirror` packages, i.e. `prosemirror-model`
> * ProseMirror type definitions, i.e. `@types/prosemirror-model`
>
> Construct a command with your package manager to remove all these dependencies
>
> Example `npm remove remirror @remirror/pm @remirror/react @remirror/extension-react-tables jest-remirror @types/prosemirror-model`
>
> When this is complete, search your **lockfile** for any remaining Remirror or ProseMirror dependencies. If any are present, you will need to determine what is requiring them.
>
> Re-add the dependencies (**excluding** the ProseMirror type definitions i.e. `@types/prosemirror-model`). 
> * Use your notes to determine which are dependencies and which are dev dependencies. Construct the commands appropriately
> 
> Example:
> `npm install remirror @remirror/react @remirror/pm @remirror/extension-react-tables`
> `npm install -D jest-remirror`

Reply:
Good point, @whawker !

Shouldn't the following also work?
```
npm install remirror @remirror/react@latest @remirror/pm@latest @remirror/extension-react-tables@latest
npm install -D jest-remirror@latest
```

Reply:
It _should_ work, but in practice I have found the complete remove, verify, and reinstall to be much more reliable. 

The method you suggested sometimes results in duplicate prosemirror-model versions etc. Sometimes dedupe helps resolve this, but not always.